src/components/Dashboard /StudentData/index.js

```diff
@@ -0,0 +1,134 @@
+      });
+  }, []);
+
+  const handleSearchChange = (e) => {
+    e.preventDefault();
+    setSearchTerm(e.target.value);
+  };
+
+  return (
+    <>
+      <div className="table-search">
+        <input
+          type="text"
+          placeholder="Search..."
+          onChange={handleSearchChange}
+          value={searchTerm}
+        />
+      </div>
+      <table className="student-overview-table">
+        <thead>
+          <tr>
+            <th>Students Name</th>
+            <th>Enrolled On </th>
+            <th>Number of class attend</th>
+            <th>Last class title</th>
+            <th>Last class attended </th>
+          </tr>
+        </thead>
+
+        {students.length > 0 &&
+          students
+            .filter((searchValue) => {
+              if (searchTerm == "") {
+                return searchValue;
+              } else if (
+                searchValue.name
+                  .toLowerCase()
+                  .includes(searchTerm.toLowerCase())
+              ) {
+                return searchValue;
+              }
+            })
+
+            .map((item) => {
+              const classEnrollDate =
+                item.created_at && item.created_at.replace("Z", "");
+
+              return (
+                <tr key={item.id}>
+                  <td>
+                    <Link
+                      style={{ textDecoration: "none" }}
+                      to={{
+                        pathname: "/student",
+                        state: { pass: item.classes_registered },
+                      }}
+                    >
+                      <td className="T-data" data-column="Students Name">
+                        {item.name}
+                      </td>
+                    </Link>
+                  </td>
+                  <td data-column="Enrolled On">
+                    {moment(classEnrollDate).format("DD-MM-YYYY")}
+                  </td>
+                  <td data-column="classes attend">
+                    {" "}
+                    {item.classes_registered.length}
+                  </td>
+
+                  <td>
+                    {item.classes_registered &&
+                    item.classes_registered.length > 0 &&
+                    item.classes_registered[item.classes_registered.length - 1][
+                      "title"
+                    ] != ""
+                      ? item.classes_registered[
+                          item.classes_registered.length - 1
+                        ]["title"]
+                      : "NA"}
+                  </td>
+                  <td>
+                    {item.classes_registered &&
+                    item.classes_registered.length > 0 &&
+                    moment(
+                      item.classes_registered[
+                        item.classes_registered.length - 1
+                      ]["end_time"].replace("Z", "")
+                    ).format("DD-MM-YYYY") != ""
+                      ? moment(
```

Comment:
do these modifications of data one time in the `then` handler of the api call. Right now, this is being calculated on every render.

Reply:
Done